AlphaWallet/TokenScriptClient/Models/XMLHandler.swift

```diff
@@ -810,9 +810,12 @@ extension XMLHandler {
         }
     }
 
-    static func getBitMask(fromTokenIdElement tokenIdElement: XMLElement) -> BigUInt? {
-        guard let bitmask = tokenIdElement["bitmask"] else { return nil }
-        return BigUInt(bitmask, radix: 16)
+    static func getBitMask(fromTokenIdElement tokenIdElement: XMLElement) -> BigUInt {
```

Comment:
Nice!

How about keeping `getBitMask()` as returning `BigUInt?` (i.e. optional), as well as storing the bitmask as an optional in the various origins types (e.g. `FunctionOrigin`), and then when applying, we can check if the bitmask is nil, just return `value` instead of "`value AND bitmask` + shift" (actually I thought that's what we did). This is better because:

1. we potentially skip an `AND` operation for every attribute with the initial load and every refresh.
2. we don't have to hardcode a constant (bitmask)
3. Being nil in the origin types (`FunctionOrigin`) etc also matches the schema where bitmask is optional, although having a default value helps with that too, but refer to [2].

If we have to add a constant for the bitmask, it'll fit better in `TokenScript.swift` because it's better to think of the TokenScript client as a separate framework/pod, which we should probably work on at some point. We might have a single cross-platform code base for the meat of TokenScript, but we'll still have quite a bit of platform-specific TokenScript client code. This is minor and easy to discover and refactor later of course.

Reply:
@hboon see latest commit. A default must be applied in the origins otherwise it will return nil and never resolve the value. I put this as a constant in TokenScript.swift and only use it in Origins init